lib/active_model_serializers/test/schema.rb

```diff
@@ -0,0 +1,77 @@
+require 'json_schema'
+
+module ActiveModelSerializers
+  module Test
+    module Schema
+      def assert_response_schema(schema_path = nil, message = nil)
+        matcher = AssertResponseSchema.new(schema_path, response, message)
+        assert(matcher.call, matcher.message)
+      end
+
+      class AssertResponseSchema
+        attr_reader :schema_path, :response, :message
+
+        def initialize(schema_path, response, message)
+          @response = response
+          @schema_path = schema_path || schema_path_default
+          @message = message
+          @document_store = JsonSchema::DocumentStore.new
+          add_schema_to_document_store
+        end
+
+        def call
+          json_schema.expand_references!(store: document_store)
+          status, errors = json_schema.validate(response_body)
+          @message ||= errors.map(&:to_s).to_sentence
+          status
+        end
+
+        private
+
+        ActiveModelSerializers.silence_warnings do
+          attr_reader :document_store
+        end
```

Comment:
this silence isn't necessary if the reader is in the protected scope, so that's what I've been doing lately. (it's a ruby bug)